### ISSUE TITLE
False positive error when evaluating generic class or type alias with TypeVarTuple followed by ParamSpec with default

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7444]
PEP 696 indicates that the following should be OK, but pyright generates an error in this case.

```python
from typing import Generic, TypeVarTuple, assert_type
from typing_extensions import ParamSpec

P = ParamSpec("P", default=[float, bool])
Ts = TypeVarTuple("Ts")

class Foo6(Generic[*Ts, P]): ...  # OK

assert_type(Foo6[int, str], type[Foo6[int, str, [float, bool]]])
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.354, which I just published.
